Convert Meijer G-function and better support for Singular Points (regular).

shubhamtibra opened the pull request.

shubhamtibra committed:
a rough code for fobrenius method

jksuom commented on sympy/holonomic/holonomic.py:
```diff
@@ -1178,6 +1178,156 @@ def to_sequence(self, lb=True):
 
         return HolonomicSequence(sol, u0)
 
+    def _fobrenius(self, indicialroots, lb=True):
```
The name is spelled 'Frobenius'.

shubhamtibra committed:
added functionality

shubhamtibra committed:
bug fix in series

shubhamtibra committed:
changes in docstring

shubhamtibra committed:
a basic method for converting to meijerg

shubhamtibra committed:
added docstring

shubhamtibra committed:
added functionality for algebraic functions and support for initial conditions in regular singular points

shubhamtibra committed:
added some tests

shubhamtibra committed:
return the expanded polynomial if given hypergeometric function represents a polynomial

shubhamtibra committed:
1. Methods `to_sequence()`, `series()`, `to_hyper()`, `to_expr()` and `to_meijerg()` now
returns a general solution if none of the indicial roots differ by an integer.

2. Better support for initial conditions in regular singular points. Users can provide initial terms
of the power series corresponding to each root of the indicial equation.

3. To meijerg returns polynomial if the associated hypergeometric function is a polynomial.

shubhamtibra committed:
fix build errors

shubhamtibra committed:
added docstrings for singular ics

shubhamtibra committed:
fixed more errors

shubhamtibra committed:
sort the roots

jksuom commented:
It seems that only real roots are sorted. What should be done with complex roots?

shubhamtibra committed:
considering complex roots if none of the indicial roots differ by an integer

jksuom commented on sympy/holonomic/holonomic.py:
```diff
@@ -1215,9 +1215,17 @@ def _frobenius(self, lb=True):
+                compl.extend([(i, a, b)] * indicialroots[i])
+        compl.sort(key=lambda x : x[1])
+        for i in range(0, len(compl)-1):
+            if compl[i][2] > 0 and compl[i][1] == compl[i+1][1]:
```
There may be several roots having the same real part. If the coefficients of the indicial equation are not real, the roots are not necessarily conjugates of each other.

shubhamtibra replied:
That is correct. I have removed this assumption.

shubhamtibra committed:
added few tests and comments

shubhamtibra committed:
change the ordering of complex roots

shubhamtibra commented:
Ping @jksuom @certik .

certik commented:
That looks good to me. Great work.
